node/src/config.rs

```diff
@@ -61,11 +61,14 @@ impl MpcConfig {
     pub fn from_participants_with_near_account_id(
         participants: ParticipantsConfig,
         my_near_account_id: &AccountId,
+        my_p2p_public_key: &ed25519_dalek::VerifyingKey,
     ) -> Option<Self> {
         let my_participant_id = participants
             .participants
             .iter()
-            .find(|p| &p.near_account_id == my_near_account_id)?
+            .find(|p| {
+                &p.near_account_id == my_near_account_id && &p.p2p_public_key == my_p2p_public_key
+            })?
```

Comment:
So the purpose of this is that if our node configuration is not present in the participant config, the node will halt?

Shouldn't we also check the address and port here? Or is that just a practical limitation since the MPC node is unaware of this information?

Reply:
> Shouldn't we also check the address and port here? 

I don't think that is necessary. When all nodes run inside a TEE, then the TLS key effectively acts as a unique identifier for the nodes.

Reply:
> When all nodes run inside a TEE, then the TLS key effectively acts as a unique identifier for the nodes.

Right because it's generated inside the TEE so you can't reuse it across instances. Fair point.